src/components/Editor/index.js

```diff
@@ -1,264 +1,69 @@
-
-    return this._plain
+    return null;
   }
 
-  recordChange = (plain, selection) => {
-    if (plain === this.undoStack[this.undoStack.length - 1]) {
-      return
-    }
-
-    if (this.undoOffset > 0) {
-      this.undoStack = this.undoStack.slice(0, -this.undoOffset)
-      this.undoOffset = 0
-    }
-
-    const timestamp = Date.now()
-    const record = { plain, selection }
-
-    // Overwrite last record if threshold is not crossed
-    if (timestamp - this.undoTimestamp < 3000) {
-      this.undoStack[this.undoStack.length - 1] = record
-    } else {
-      this.undoStack.push(record)
-
-      if (this.undoStack.length > 50) {
-        this.undoStack.shift()
-      }
-    }
-
-    this.undoTimestamp = timestamp
-  }
-
-  updateContent = plain => {
-    if (this.compositing) {
-      return;
-    }
-    this.setState({ html: prism(plain, this.props.language) })
-
-    if (this.props.onChange) {
-      this.props.onChange(plain)
-    }
-  }
-
-  restoreStackState = offset => {
-    const { plain, selection } = this.undoStack[this.undoStack.length - 1 - offset]
-
-    this.selection = selection
-    this.undoOffset = offset
-    this.updateContent(plain)
-  }
-
-  undo = () => {
-    const offset = this.undoOffset + 1
-    if (offset >= this.undoStack.length) {
-      return
-    }
-
-    this.restoreStackState(offset)
-  }
-
-  redo = () => {
-    const offset = this.undoOffset - 1
-    if (offset < 0) {
-      return
-    }
-
-    this.restoreStackState(offset)
-  }
-
-  onKeyDown = evt => {
-    if (this.props.onKeyDown) {
-      this.props.onKeyDown(evt)
-    }
-
-    if (evt.keyCode === 9 && !this.props.ignoreTabKey) { // Tab Key
-      document.execCommand('insertHTML', false, '  ')
-      evt.preventDefault()
-    } else if (evt.keyCode === 8) { // Backspace Key
-      const { start: cursorPos, end: cursorEndPos } = selectionRange(this.ref)
-      if (cursorPos !== cursorEndPos) {
-        return // Bail on selections
-      }
-
-      const deindent = getDeindentLevel(this.getPlain(), cursorPos)
-      if (deindent <= 0) {
-        return // Bail when deindent level defaults to 0
-      }
-
-      // Delete chars `deindent` times
-      for (let i = 0; i < deindent; i++) {
-        document.execCommand('delete', false)
-      }
+  state = {
+    code: ''
+  };
 
-      evt.preventDefault()
-    } else if (evt.keyCode === 13) { // Enter Key
-      const { start: cursorPos } = selectionRange(this.ref)
-      const indentation = getIndent(this.getPlain(), cursorPos)
-      document.execCommand('insertHTML', false, '\n' + indentation)
-      evt.preventDefault()
-    } else if (
-      // Undo / Redo
-      evt.keyCode === 90 &&
-      evt.metaKey !== evt.ctrlKey &&
-      !evt.altKey
-    ) {
-      if (evt.shiftKey) {
-        this.redo()
-      } else {
-        this.undo()
+  updateContent = code => {
+    this.setState({ code }, () => {
+      if (this.props.onChange && typeof this.props.onChange === 'function') {
```

Comment:
Is there ever a case where `typeof this.props.onChange` is not a function? Given the PropTypes below I would think this condition would never be challenged